lib/improver/utilities/cube_manipulation.py

```diff
@@ -193,12 +188,380 @@ def concatenate_cubes(
+
+    Args:
+        cubes : Iris cubelist
+            List of cubes to compare (must be more than 1)
+
+    Returns:
+        common_keys : List
+            List of common attribute keys (str)
+        unmatching_attribues : List
+            List of dictionaries of unmatching attributes
+
+    Warns:
+        Warning: If only a single cube is supplied
+    """
+    common_keys = list(cubes[0].attributes.keys())
+    unmatching_attributes = []
+    if isinstance(cubes, iris.cube.Cube) or len(cubes) == 1:
+        msg = ('Only a single cube so no differences will be found ')
+        warnings.warn(msg)
+    else:
+        for cube in cubes[1:]:
+            cube_keys = list(cube.attributes.keys())
+            common_keys = [
+                key for key in common_keys
+                if (key in cube_keys and
+                    np.all(cube.attributes[key] == cubes[0].attributes[key]))]
+
+        for i, cube in enumerate(cubes):
+            unmatching_attributes.append(dict())
+            for key in list(cube.attributes.keys()):
+                if key not in common_keys:
+                    unmatching_attributes[i].update({key:
+                                                     cube.attributes[key]})
+    return common_keys, unmatching_attributes
+
+
+def compare_coords(cubes):
+    """
+    Function to compare attributes of cubes
+
+    Args:
+        cubes : Iris cubelist
+            List of cubes to compare (must be more than 1)
+
+    Returns:
+        common_coords : List
+            List of common coordinates (iris.Coord)
+        unmatching_coords : List
+            List of dictionaries of unmatching coordinates
+
+    Warns:
+        Warning: If only a single cube is supplied
+    """
+    common_coords = list(cubes[0].coords())
+    unmatching_coords = []
+    if isinstance(cubes, iris.cube.Cube) or len(cubes) == 1:
+        msg = ('Only a single cube so no differences will be found ')
+        warnings.warn(msg)
+    else:
+        for cube in cubes[1:]:
+            cube_coords = list(cube.coords())
+            common_coords = [
+                coord for coord in common_coords
+                if (coord in cube_coords and
+                    np.all(cube.coords(coord) == cubes[0].coords(coord)))]
+
+        for i, cube in enumerate(cubes):
+            unmatching_coords.append(dict())
+            for coord in list(cube.coords()):
+                if coord not in common_coords:
+                    dim_coords = cube.dim_coords
+                    if coord in dim_coords:
+                        dim_val = dim_coords.index(coord)
+                    else:
+                        dim_val = -1
+                    unmatching_coords[i].update({coord.name():
+                                                 {'data_dims': dim_val,
+                                                  'coord': coord}})
+
+    return common_coords, unmatching_coords
+
+
+def build_coordinate(data, long_name=None,
```

Comment:
How similar is this to the build_coordinate functionality in spotdata/extract_data.py? Do we need to consolidate these utilities?